lib/environment/permissions.js

```diff
@@ -9,52 +9,44 @@ function filterPerms(perms) {
 	return { permissions, origins };
 }
 
-const STORAGE_KEY = 'RES.cachedDeniedPerms';
+export const has = _.memoize(
+	async (...perms) => {
```

Comment:
Nit: you can remove `async` and return the promise directly.

Also, I think all callsites now spread the array, so you could replace the `...perms` rest param with a single param taking an array.